packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/api/networkAnalyticsContext.ts

```diff
@@ -2,30 +2,53 @@
 // Licensed under the MIT license.
 
 import { TokenCredential } from "@azure/core-auth";
-import { ClientOptions } from "@azure-rest/core-client";
-import { NetworkAnalyticsContext } from "../rest/index.js";
-import getClient from "../rest/index.js";
+import { ClientOptions, Client, getClient } from "@azure-rest/core-client";
+import { logger } from "../logger.js";
+
+export interface NetworkAnalyticsContext extends Client {}
 
 /** Optional parameters for the client. */
 export interface NetworkAnalyticsClientOptionalParams extends ClientOptions {
   /** The API version to use for this operation. */
   apiVersion?: string;
 }
 
-export { NetworkAnalyticsContext } from "../rest/index.js";
-
 export function createNetworkAnalytics(
   credential: TokenCredential,
   options: NetworkAnalyticsClientOptionalParams = {},
 ): NetworkAnalyticsContext {
+  const endpointUrl =
+    options.endpoint ?? options.baseUrl ?? `https://management.azure.com`;
+
   const prefixFromOptions = options?.userAgentOptions?.userAgentPrefix;
   const userAgentPrefix = prefixFromOptions
     ? `${prefixFromOptions} azsdk-js-api`
     : "azsdk-js-api";
-
-  const clientContext = getClient(credential, {
+  const updatedOptions = {
     ...options,
     userAgentOptions: { userAgentPrefix },
+    loggingOptions: { logger: options.loggingOptions?.logger ?? logger.info },
+    credentials: {
+      scopes: options.credentials?.scopes ?? [`${endpointUrl}/.default`],
+    },
+  };
+  const clientContext = getClient(endpointUrl, credential, updatedOptions);
+  clientContext.pipeline.removePolicy({ name: "ApiVersionPolicy" });
+  const apiVersion = options.apiVersion ?? "2023-11-15";
```

Comment:
@wanlwanl I thnk we need to change the version detection logic accordingly, after this pr get merged from `src/rest/XXXClient.ts` to `src/api/XXXContext.ts`